### DISCUSSION TITLE
partial biuld

### DISCUSSION THREAD

**Discussion by MattiaGrigoli** [Joshua-Riek/ubuntu-rockchip#1109] [Q&A]
is there a way to do a partial build of the image? (only kernel, only u-boot...)
because i'm creating a custom image and for rebuilding everything it takes 1 to 2 hours every time, so 'i'm looking for a way to shorten this time by only building the modified stuff and keeping the ones not modified.

**Comment by hakotani-o** [comment]
Hi MattiaGrigoli.

Please see build.sh.

usage() {
cat << HEREDOC
Usage: $0 --board=[orangepi-5] --suite=[jammy|noble] --flavor=[server|desktop]

Required arguments:
  -b, --board=BOARD      target board 
  -s, --suite=SUITE      ubuntu suite 
  -f, --flavor=FLAVOR    ubuntu flavor

Optional arguments:
  -h,  --help            show this help message and exit
  -c,  --clean           clean the build directory
  -ko, --kernel-only     only compile the kernel
  -uo, --uboot-only      only compile uboot
  -ro, --rootfs-only     only build rootfs
  -l,  --launchpad       use kernel and uboot from launchpad repo
  -v,  --verbose         increase the verbosity of the bash script
HEREDOC
}

rootfs can download from Actions -> Nightly build -> Build rootfs.

Thank you.
